extension-manifest-v3/src/utils/engines.js

```diff
@@ -123,15 +125,28 @@ async function loadFromStorage(name) {
 
 async function saveToStorage(name) {
   const engine = loadFromMemory(name);
-  const db = await getDB();
+  const serialized = engine?.serialize();
 
-  const tx = db.transaction('engines', 'readwrite');
-  const table = tx.objectStore('engines');
+  try {
+    const db = await getDB();
 
-  if (engine) {
-    await table.put(engine.serialize(), name);
-  } else {
-    await table.delete(name);
+    const tx = db.transaction('engines', 'readwrite');
+    const table = tx.objectStore('engines');
+
+    if (engine) {
+      await table.put(serialized, name);
+    } else {
+      await table.delete(name);
+    }
```

Comment:
At this point, we have successfully updated IndexedDB. It is the source of truth, and I assume it is likely that it could be read on the startup.

So, I wonder if we should do a best-effort (fire-and-forget operation) on Firefox here to purge the fallback in chrome.storage.local?

Reply:
This is valid question, but we would need to read storage for 99% users, which don't need it. You cannot know if the "bug" state was, it is not here, and we should clean up the storage.

Reply:
That is true. Might not be the best place to do it here. I'm fine with leaving that out of the PR here.

Leaving old engines around is probably not harmful. They have some space overhead, but I don't think it results in inconsistencies. For problems like that, we could think of having a different place in the extension (e.g. during an extension update when migrations happen), where we could put cleanup. Keeping timestamps is generally useful, because then it possible to detect data that has expired.

Reply:
As said, I'm fine with leaving it out. Though when I think about it, the overhead of a delete to storage should not be a issue, since it happens in a path where we update the engine. This is an expensive IO write operation already. In that path, having an extra delete of a value that will almost certainly not exist, is avoidable overhead for most Firefox users, but probably not noticeable.

Reply:
I thought that you meant the `loadFromStorage`. You're right, as the saveToStorage is not that often, I can add it there.